app/models/infra_conversion_job.rb

```diff
@@ -0,0 +1,160 @@
+class InfraConversionJob < Job
+  POLL_CONVERSION_INTERVAL = 60
+  POLL_CONVERSION_MAX = 24 * 60 # i.e. default 24 hour
+  POLL_POST_STAGE_MAX = 30 # i.e. default 30 minutes
+
+  def self.create_job(options)
+    # TODO: from settings/user plan settings
+    options[:conversion_polling_interval] = POLL_CONVERSION_INTERVAL
+    options[:poll_conversion_max] = POLL_CONVERSION_MAX
+    options[:poll_post_stage_max] = POLL_POST_STAGE_MAX
+    super(name, options)
+  end
+
+  #
+  # State-transition diagram:
+  #                              :poll_conversion                         :poll_post_stage
+  #    *                          /-------------\                        /---------------\
+  #    | :initialize              |             |                        |               |
+  #    v               :start     v             |                        v               |
+  # waiting_to_start --------> running ------------------------------> post_conversion --/
+  #     |                         |                :start_post_stage       |
+  #     | :abort_job              | :abort_job                             |
+  #     \------------------------>|                                        | :finish
+  #                               v                                        |
+  #                             aborting --------------------------------->|
+  #                                                    :finish             v
+  #                                                                    finished
+  #
+
+  alias_method :initializing, :dispatch_start
+  alias_method :finish,       :process_finished
+  alias_method :abort_job,    :process_abort
+  alias_method :cancel,       :process_cancel
+  alias_method :error,        :process_error
+
+  def load_transitions
+    self.state ||= 'initialize'
+
+    {
+      :initializing     => {'initialize'       => 'waiting_to_start'},
+      :start            => {'waiting_to_start' => 'running'},
+      :poll_conversion  => {'running'          => 'running'},
+      :start_post_stage => {'running'          => 'post_conversion'},
+      :poll_post_stage  => {'post_conversion'  => 'post_conversion'},
+      :finish           => {'*'                => 'finished'},
+      :abort_job        => {'*'                => 'aborting'},
+      :cancel           => {'*'                => 'canceling'},
+      :error            => {'*'                => '*'}
+    }
+  end
+
+  def migration_task
+    @migration_task ||= target_entity
```

Comment:
The target_entity is known at initialization time right?  This looks like it could just be an `attr_reader` that is set in initialize

Reply:
This is caching the [query result of `target_entity`](https://github.com/jameswnl/manageiq/blob/throttle/app/models/job.rb#L150).  Shall I rely on rails to take care of caching, and just do a `alias_method :migration_task, :target_entity`?

Reply:
Oh no problem with the caching, I was saying in `initialize` you could do `@migration_task = target_entity` and make `migration_task` an `attr_reader` which is basically what this duplicates the behavior of.